app/models/service.rb

```diff
@@ -137,6 +137,14 @@ def power_status
     options[:power_status]
   end
 
+  def self.active
+    Service.where(:lifecycle_state => ["provisioning", "provisioned"], :retired => false)
+  end
+
+  def self.inactive
+    Service.where(:lifecycle_state => ["unprovisioned", "error_in_provisioning"], :retired => false) + Service.retired
+  end
```

Comment:
This is strange because there is no relation between the lifecycle states and whether a service is marked as retired or not. That means you can have any lifecycle state whether the service is marked as retired or not

Reply:
That's really weird :(